beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -4427,14 +4189,12 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             self.block_times_cache.write().prune(slot);
 
             // Don't run heavy-weight tasks during sync.
-            if self.best_slot().map_or(true, |head_slot| {
-                head_slot + MAX_PER_SLOT_FORK_CHOICE_DISTANCE < slot
-            }) {
+            if self.best_slot() + MAX_PER_SLOT_FORK_CHOICE_DISTANCE < slot {
                 return;
             }
 
             // Run fork choice and signal to any waiting task that it has completed.
```

Comment:
We signal the fork choice notifier from a blocking task when we signal it from the state advance, seems we may as well do the same here (there's a `notify_fork_choice_complete` on line 4209).

Reply:
Could you please confirm this is what you were intending? :pray: 

d14ffd478

Reply:
Oh yeah, that's not what I meant sorry. I meant we should use a spawn blocking to do the notification

Reply:
I think https://github.com/sigp/lighthouse/commit/d14ffd478bd50dabca9b1fc152d9d8c2ecf83bc6 should be reverted too

Reply:
Ooh yep, I'm with you sorry! How's this one? cfd9a7f77a81b3a03bf5acf8811807eac465c49c

Reply:
Looking good!